### PR TITLE
Minimal support for MIME secret encoding 

### PR DESCRIPTION
**Pull request by minjatJ**
Hi @msfjarvis ,

First all thanks a lot for this application, I have been using it for a long it and it has always served me well. 
This PR should provide the minimum support required for https://github.com/android-password-store/Android-Password-Store/issues/1055
To be honest I'm quite saddened by Gopass teams' decision of breaking compatibility with pass but in the meantime I really like their tool.
I understand you don't want to "pollute" your solution with whatever crazy format everybody come up with but as **username** is already retrieved from *extraContext* in a key/value fashion I thought **password** can be retrieved the same way without too much (any?) breaking changes and bringing minimum compatibility with Gopass 1.9+ new secret format.

## :loudspeaker: Type of change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## :scroll: Description

The **password** field in `PasswordEntry` model was set to the first line of the secret content (Unless `UriTotpFinder` was found).
This PR brings an extra step where if no `UriTotpFinder` is found the **password** is retrieved from `passContent` in a key/value fashion (e.g: "password: mySecretPassword") and fallback to the first line of the `passContent` if nothing is found.
Also `passContent` is transformed to a **List<String>** so `findAndStripPassword` can alter and return it (And remove the line containing our password from it) so then **extraContent** contains the whole extra content without our password line.

## :bulb: Motivation and Context

This PR should provide the minimum support required for https://github.com/android-password-store/Android-Password-Store/issues/1055

## :green_heart: How did you test it?
- Android `Run All Tests`
- Tested on my android phone using my personal password repository managed by Gopass 1.10+

## :pencil: Checklist

- [X] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [X] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

## :camera_flash: Screenshots / GIFs

PS: This is my first time writing Kotlin so please advise on any best practice I have missed.

### CODE REVIEW

**Comment by msfjarvis** [comment]
I'd like to see [some tests](https://github.com/android-password-store/Android-Password-Store/blob/develop/app/src/test/java/com/zeapo/pwdstore/model/PasswordEntryTest.kt) for this, there are many examples in the existing tests on how to do it.

**Review by fmeum — COMMENTED** [review]
Thanks for the PR, this already looks pretty good!

Just one recommendation: It is usually better to use immutable collections if possible as they increase the risk of introducing subtle bugs. 
How about turning `findPassword` into a function `findAndStripPassword` that returns `Pair<String,List<String>>`, where the first element of the pair is the password and the second contains the remaining lines?

**Comment by minjatJ** [comment]
There you go! For some reason I wasn't able to use
```kotlin
val passContent = content.split("\n".toRegex())
(password, passContent) = findAndStripPassword(passContent)
```
Is it just because `passContent` can't be re-assigned ? Is there a way to flatten a pair in one line?
Thanks!

**Comment by fmeum** [comment]
> There you go! For some reason I wasn't able to use
> ```kotlin
> val passContent = content.split("\n".toRegex())
> (password, passContent) = findAndStripPassword(passContent)
> ```
> Is it just because `passContent` can't be re-assigned ? Is there a way to flatten a pair in one line?
> Thanks!

You should be able to do
```
val (password, newPassContent) = ...
```

**Comment by minjatJ** [comment]
> > There you go! For some reason I wasn't able to use
> > ```kotlin
> > val passContent = content.split("\n".toRegex())
> > (password, passContent) = findAndStripPassword(passContent)
> > ```
> > 
> > 
> > Is it just because `passContent` can't be re-assigned ? Is there a way to flatten a pair in one line?
> > Thanks!
> 
> You should be able to do
> 
> ```
> val (password, newPassContent) = ...
> ```

Won't it create a new variable for `password` and not use the one from the object we initialize?

File: app/src/main/java/com/zeapo/pwdstore/model/PasswordEntry.kt [hunk 1]
```diff
@@ -31,9 +31,9 @@ class PasswordEntry(content: String, private val totpFinder: TotpFinder = UriTot
     constructor(os: ByteArrayOutputStream) : this(os.toString("UTF-8"), UriTotpFinder())
 
     init {
-        val passContent = content.split("\n".toRegex(), 2).toTypedArray()
-        password = if (UriTotpFinder.TOTP_FIELDS.any { passContent[0].startsWith(it) }) "" else passContent[0]
-        extraContent = findExtraContent(passContent)
+        val pair = findAndStripPassword(content.split("\n".toRegex()))
+        password = pair.first
+        extraContent = pair.second.joinToString("\n")
```
Comment (minjatJ) [inline app/src/main/java/com/zeapo/pwdstore/model/PasswordEntry.kt]: I can change by the following
```suggestion
        val (foundPassword, passContent) = findAndStripPassword(content.split("\n".toRegex()))
        password = foundPassword
        extraContent = passContent.joinToString("\n")
```

File: app/src/main/java/com/zeapo/pwdstore/model/PasswordEntry.kt [hunk 2]
```diff
@@ -86,10 +86,16 @@ class PasswordEntry(content: String, private val totpFinder: TotpFinder = UriTot
         return null
     }
 
-    private fun findExtraContent(passContent: Array<String>) = when {
-        password.isEmpty() && passContent[0].isNotEmpty() -> passContent[0]
-        passContent.size > 1 -> passContent[1]
-        else -> ""
+    private fun findAndStripPassword(passContent: List<String>): Pair<String, List<String>> {
+        if (UriTotpFinder.TOTP_FIELDS.any { passContent[0].startsWith(it) }) return Pair("", passContent)
+        for ((index, line) in passContent.withIndex()) {
```
Comment (fmeum) [inline app/src/main/java/com/zeapo/pwdstore/model/PasswordEntry.kt]:
```suggestion
        for (line in passContent) {
```

File: app/src/main/java/com/zeapo/pwdstore/model/PasswordEntry.kt (hunk 1 above)
Comment (fmeum) [reply app/src/main/java/com/zeapo/pwdstore/model/PasswordEntry.kt]: This looks good.

**Review by fmeum — DISMISSED** [review]
Looks good to me now. Thanks! @msfjarvis Could you sign off on this?

I do think that a major refactor of our password entry parsing is due and know at least a few things I'll improve in an upcoming commit, but this could go in as is (with the two suggested changes).

**Review by msfjarvis — APPROVED** [review]

**Comment by msfjarvis** [comment]
Thanks for your contribution! I'll include this in the v1.13.1 release that'll be out in a couple hours.

**Pull request merged**